transport-native-unix-common/src/main/java/io/netty/channel/unix/PreferredDirectByteBufAllocator.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.unix;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.util.internal.UnstableApi;
+
+@UnstableApi
+public final class PreferredDirectByteBufAllocator implements ByteBufAllocator {
```

Comment:
This was needed as otherwise we may fail to read data if a `ByteBufAllocator` is used which returns heap buffers from `ioBuffer(...)` methods.